### PR TITLE
[sipify] handle python code snippets

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

File: python/core/geometry/qgsabstractgeometry.sip [hunk 1]
```diff
@@ -458,7 +458,7 @@ It can also be thought as rounding the edges and it may be useful for removing e
 Example:
 \code
 geometry->snappedToGrid(1, 1);
-\endcode
```
Comment (nyalldawson) [inline python/core/geometry/qgsabstractgeometry.sip]: What's happened here?

File: python/core/qgscoordinatereferencesystem.sip [hunk 2]
```diff
@@ -44,21 +44,21 @@ Transformations between coordinate reference systems are done using QgsCoordinat
...
+     else:
+         print("Invalid CRS!")
 
 This will produce the following output:
 
-~~~
-CRS Description: OSGB 1936 / British National Grid
-CRS PROJ.4 text: +proj=tmerc +lat_0=49 +lon_0=-2 +k=0.9996012717 +x_0=400000 +y_0=-100000 [output trimmed]
-~~~
+.. code-block:: python
```
Comment (nyalldawson) [inline python/core/qgscoordinatereferencesystem.sip]: Not python ;)

File: python/core/expression/qgsexpression.sip [hunk 3]
```diff
@@ -18,22 +18,22 @@ Class for parsing and evaluation of expressions (formerly called "search strings
...
-if (exp.hasEvalError())
-{
-// show error message with evalErrorString()
-}
-else
-{
-// examine the result
-}
-\endcode
+.. code-block:: cpp
```
Comment (nyalldawson) [inline python/core/expression/qgsexpression.sip]: Should non python examples be hidden?

Comment (3nids) [reply python/core/expression/qgsexpression.sip]: fixed

File: python/core/geometry/qgsabstractgeometry.sip (hunk 1 above)
Comment (3nids) [reply python/core/geometry/qgsabstractgeometry.sip]: fixed

**Pull request merged**